test/document_stores/test_opensearch.py

```diff
@@ -1,12 +1,706 @@
+        _init_client_params["aws4auth"] = None
+        OpenSearchDocumentStore._init_client(**_init_client_params)
+        _, kwargs = mocked_open_search_init.call_args
+        assert "http_auth" not in kwargs
+
+    def test_query_by_embedding_raises_if_missing_field(self, mocked_document_store):
+        mocked_document_store.embedding_field = ""
+        with pytest.raises(RuntimeError):
+            mocked_document_store.query_by_embedding(self.query_emb)
+
+    def test_query_by_embedding_filters(self, mocked_document_store):
+        expected_filters = {"type": "article", "date": {"$gte": "2015-01-01", "$lt": "2021-01-01"}}
+        mocked_document_store.query_by_embedding(self.query_emb, filters=expected_filters)
+        # Assert the `search` method on the client was called with the filters we provided
+        _, kwargs = mocked_document_store.client.search.call_args
+        actual_filters = kwargs["body"]["query"]["bool"]["filter"]
+        assert actual_filters["bool"]["must"] == [
+            {"term": {"type": "article"}},
+            {"range": {"date": {"gte": "2015-01-01", "lt": "2021-01-01"}}},
+        ]
+
+    def test_query_by_embedding_return_embedding_false(self, mocked_document_store):
+        mocked_document_store.return_embedding = False
+        mocked_document_store.query_by_embedding(self.query_emb)
+        # assert the resulting body is consistent with the `excluded_meta_data` value
+        _, kwargs = mocked_document_store.client.search.call_args
+        assert kwargs["body"]["_source"] == {"excludes": ["embedding"]}
+
+    def test_query_by_embedding_excluded_meta_data_return_embedding_true(self, mocked_document_store):
+        """
+        Test that when `return_embedding==True` the field should NOT be excluded even if it
+        was added to `excluded_meta_data`
+        """
+        mocked_document_store.return_embedding = True
+        mocked_document_store.excluded_meta_data = ["foo", "embedding"]
+        mocked_document_store.query_by_embedding(self.query_emb)
+        _, kwargs = mocked_document_store.client.search.call_args
+        # we expect "embedding" was removed from the final query
+        assert kwargs["body"]["_source"] == {"excludes": ["foo"]}
+
+    def test_query_by_embedding_excluded_meta_data_return_embedding_false(self, mocked_document_store):
+        """
+        Test that when `return_embedding==False`, the final query excludes the `embedding` field
+        even if it wasn't explicitly added to `excluded_meta_data`
+        """
+        mocked_document_store.return_embedding = False
+        mocked_document_store.excluded_meta_data = ["foo"]
+        mocked_document_store.query_by_embedding(self.query_emb)
+        # assert the resulting body is consistent with the `excluded_meta_data` value
+        _, kwargs = mocked_document_store.client.search.call_args
+        assert kwargs["body"]["_source"] == {"excludes": ["foo", "embedding"]}
```

Comment:
No warnings are emitted. I am hesitant adding warnings because I am not sure what should be the right precedence order and there isn't an obvious answer. There are several parts in this document store where intents aren't clear and in some cases this results in bugs - I fixed one in this PR because it was so obvious to break multiple tests but otherwise I'd like to limit the scope to introducing the new client and tests.